specification/kubernetesruntime/KubernetesRuntime.Management/loadbalancer.tsp

```diff
@@ -0,0 +1,65 @@
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+
+namespace Microsoft.KubernetesRuntime;
+
+@doc("A LoadBalancer resource")
+model LoadBalancer is ExtensionResource<LoadBalancerProperties> {
+  @doc("The name of the LoadBalancer")
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @key("loadBalancerName")
+  @segment("loadBalancers")
+  @path
+  name: string;
+}
+
+@doc("Enum of advertise mode")
+enum AdvertiseMode {
+  @doc("ARP advertise mode")
+  ARP,
+
+  @doc("BGP advertise mode")
+  BGP,
+
+  @doc("both ARP and BGP advertise mode")
+  Both,
+}
+
+@doc("Details of the LoadBalancer.")
+model LoadBalancerProperties {
+  @doc("IP Range")
+  addresses: string[];
+
+  @doc("A dynamic label mapping to select related services. For instance, if you want to create a load balancer only for services with label \"a=b\", then please specify {\"a\": \"b\"} in the field.")
+  serviceSelector?: Record<string>;
```

Comment:
This is resulting in warnings for using additional properties... is this an existing API or a new one? Have you considered defining this as an object array instead?

Reply:
This is an existing API. It's documented that we need the dynamic dictionary structure because the label is controlled by the customers.